arrow-libs/fx/arrow-fx-stm/src/commonMain/kotlin/arrow/fx/stm/TVar.kt

```diff
@@ -196,7 +196,7 @@ public class TVar<A> internal constructor(a: A) {
    * This does not happen implicitly on [release] because release may also write the same value back on
    *  normal lock release.
    */
-  internal fun registerWaiting(trans: STMTransaction<*>, expected: A): Boolean {
+  internal fun registerWaiting(trans: STMTransaction, expected: A): Boolean {
```

Comment:
Raw types are [not super-well supported in Kotlin](https://youtrack.jetbrains.com/issue/KT-48345/Try-getting-rid-of-raw-types-in-Kotlin), so I would suggest rolling this change back to a star-projected type.

Reply:
This isn't a raw type. I changed `STMTransaction` to no longer need that type since it wasn't used anywhere. Its `commit` method now takes the function to use directly, instead of that being passed into the constructor. This changed almost none of the code because `STMTransaction` was only created in `atomically`, and then had `commit` immediately called on it.

Reply:
Aha, I didn't realize that! Thanks for the explanation :)

Reply:
Thanks for introducing me to Kotlin having raw types. I had no idea that existed at all!